src/transforms/Babel.ts

```diff
@@ -21,19 +22,27 @@ export class BabelTransform {
 
     // eslint-disable-next-line
     this.babel = require(
-      requireResolve('babel-core', { cwd: this.options.servicePath }),
+      requireResolve(this.options.babelCore, { cwd: this.options.servicePath }),
     );
   }
 
   run ({ code, map, relPath }) {
     let result = { code, map, relPath };
 
     try {
-      const transformed = this.babel.transform(code, {
-        ...this.config,
-        sourceFileName  : relPath,
-        sourceMapTarget : relPath,
-      });
+      let transformed = null;
```

Comment:
Write this like

```ts
		const sourceMapTarget = semver.gt(... blah blah) ? undefined : relPath;
        const transformed = this.babel.transform(code, {
          ...this.config,
          sourceFileName: relPath,
          sourceMapTarget,
        });

```

Reply:
Dude I gave you the code, just copy paste it, dont `delete` that is RETARDED

Reply:
@nfour your code don't works, if I applied then, I already have the field sourceMapTarget = undefined and so babel7 tell me that the field is not compliant with babel7. thats why I recode this with a delete which is the second fashion to remove a property into an object on nodejs.
Thanks

Reply:
Then define a subset of the object in the ternary instead, then spread it in eg.

```ts
const transformConfig = semver.gt(... blah blah) ? { ... }: { ... ;
const transformed = (blah blah) {
  ...this.config,
 ...transformConfig,
}
```

:)

Reply:
@nfour resolved.